nodeJS/express/views.md

```diff
@@ -78,9 +78,29 @@ Start the server and go to the `/` route in the browser. You should see:
 EJS rocks!
 ```
 
-If you inspect the HTML in the browser's dev tools, you can see HTML is structures exactly like how we wrote the EJS template with the `message` variable replaced with its value.
+If you inspect the HTML in the browser's dev tools, you can see the HTML is structured exactly like how we wrote the EJS template with the `message` variable replaced with its value.
 
-When you hit the `/` route, `res.render("index", { message: "EJS rocks!" });` is the line that sends back the response. Since we've already defined the `views` and `view engine` app properties, the first argument of `res.render` is programmed to look for "a template called index in the specified folder", while the second argument is an object of variables that are to be made available to that specific template.
+When you hit the `/` route, `res.render("index", { message: "EJS rocks!" });` is the line that sends back the response. Since we've already defined the `views` and `view engine` app properties, the first argument of `res.render` is programmed to look for "a template called index in the specified folder", while the second argument is an object of variables that are to be made available to that specific template. This object of variables is known as the `locals` object which is always generated by EJS. Say you define a variable called `foo` in the `locals` object. This variable can be accessed in the view file with either `locals.foo` or `foo`. A few more details can be found in the [Express docs](https://expressjs.com/en/5x/api.html#res.locals).
```

Comment:
A couple of bits:

- This `locals` thing is an EJS thing. [EJS docs](https://ejs.co/) - go to the `Options` section then to the `localsName` option. It's the default name for the object in EJS where locals (be that set via `res.locals` in Express, or by passing it in via the 2nd arg in `res.render). The Express docs for `res.locals` won't be relevant for this specific point. It can still be linked when discussing `res.locals`, but when discussing the EJS `locals` variable, that's an EJS-specific thing (e.g. Pug doesn't have it, and handles locals variables in a different way).
- We can split the discussion about `locals` into its own subsection with the heading `#### The locals variable in EJS`, rewording as necessary to accommodate the structure change.

Reply:
Thanks for the feedback!

Would it be worth saying that it is possible to implement local variables with both `res.locals` and the `locals` object in `res.render()`, to give the learner more options to consider? And then explicitly say that this is relevant to Express rather than EJS?

I'll add something to convey the point that the `localsName` / `locals` syntax is specific to EJS.

Reply:
No need for `app.locals`. Most people will only need `res.locals` if something like that's appropriate, and if they really need an `app.locals` thing, they can discover that as needed.

I'll address the first part in a separate review comment

Reply:
Alright, that sounds reasonable. Do I need to resolve this (and all conversations) for a PR to go through, by the way?

Reply:
Unresolved conversations won't block PRs (but even if they did, we can mark them resolved if we needed to anyway).